File: ExampleChocApp/ExampleChocApp.js
var chocOutput = {
  nodes: [
   { data: { id: 'root', name: 'root', noun: 'lightgrey' } },
 { data: { id: 'homepage', name: 'homepage', noun: 'lightgrey' } },
 { data: { id: 'categories', name: 'categories', noun: 'lightgrey' } },
 { data: { id: 'review', name: 'review', noun: 'lightgrey' } },
 { data: { id: 'reviews', name: 'reviews', noun: 'lightgrey' } },
 { data: { id: 'products', name: 'products', noun: 'lightgrey' } },
 { data: { id: 'product', name: 'product', noun: 'lightgrey' } },
 { data: { id: 'cart', name: 'cart', noun: 'lightgrey' } },
 { data: { id: 'invoice', name: 'invoice', noun: 'green' } },
 { data: { id: 'vendors', name: 'vendors', noun: 'lightgrey' } },
 { data: { id: 'vendor', name: 'vendor', noun: 'lightgrey' } },
 { data: { id: 'error', name: 'error', noun: 'red' } },

  ],
  edges: [
   { data: { source: 'root', target: 'homepage', label: 'shoppe', color: 'green' } },
 { data: { source: 'homepage', target: 'product', label: 'top_5_product_id', color: 'green' } },
 { data: { source: 'homepage', target: 'categories', label: 'by_category', color: 'green' } },
 { data: { source: 'homepage', target: 'vendors', label: 'by_vendor', color: 'green' } },
 { data: { source: 'categories', target: 'products', label: 'category_id', color: 'green' } },
 { data: { source: 'products', target: 'product', label: 'product_id', color: 'green' } },
 { data: { source: 'products', target: 'products', label: 'by_rating', color: 'green' } },
 { data: { source: 'product', target: 'reviews', label: 'reviews', color: 'green' } },
 { data: { source: 'reviews', target: 'review', label: 'review_id', color: 'green' } },
 { data: { source: 'reviews', target: 'review', label: 'add', color: 'purple' } },
 { data: { source: 'review', target: 'review', label: 'edit', color: 'blue' } },
 { data: { source: 'product', target: 'cart', label: 'add', color: 'purple' } },
 { data: { source: 'cart', target: 'cart', label: 'remove', color: 'red' } },
 { data: { source: 'cart', target: 'invoice', label: 'buy', color: 'purple' } },
 { data: { source: 'product', target: 'vendor', label: 'vendor_id', color: 'green' } },
 { data: { source: 'vendors', target: 'vendor', label: 'vendor_id', color: 'green' } },
 { data: { source: 'vendor', target: 'products', label: 'products', color: 'green' } },
 { data: { source: 'error', target: 'root', label: 'home', color: 'green' } },

  ]
};

File: ExampleChocApp/ExampleChocApp.rb
require 'sinatra'
require 'json'
require 'sinatra/json'
require_relative 'affordances/get_homepage'
require_relative 'affordances/get_product'
require_relative 'affordances/get_categories'
require_relative 'affordances/get_vendors'
require_relative 'affordances/get_products'
require_relative 'affordances/get_product'
require_relative 'affordances/get_products'
require_relative 'affordances/get_reviews'
require_relative 'affordances/get_review'
require_relative 'affordances/post_review'
require_relative 'affordances/put_review'
require_relative 'affordances/post_cart'
require_relative 'affordances/delete_cart'
require_relative 'affordances/post_invoice'
require_relative 'affordances/get_vendor'
require_relative 'affordances/get_vendor'
require_relative 'affordances/get_products'
require_relative 'affordances/get_root'


get '/homepage' do
  json(GetHomepage.new(params).response)
end

get '/products/:product_id' do
  json(GetProduct.new(params).response)
end

get '/categories' do
  json(GetCategories.new(params).response)
end

get '/vendors' do
  json(GetVendors.new(params).response)
end

get '/categories/:category_id' do
  json(GetProducts.new(params).response)
end

get '/products/:product_id' do
  json(GetProduct.new(params).response)
end

get '/products' do
  json(GetProducts.new(params).response)
end

get '/products/:product_id/reviews' do
  json(GetReviews.new(params).response)
end

get '/reviews/:review_id' do
  json(GetReview.new(params).response)
end

post '/review' do
  json(PostReview.new(params).response)
end

put '/review' do
  json(PutReview.new(params).response)
end

post '/cart' do
  json(PostCart.new(params).response)
end

delete '/cart/:cart_id/:product_id' do
  json(DeleteCart.new(params).response)
end

post '/invoice' do
  json(PostInvoice.new(params).response)
end

get '/product/:vendor_id' do
  json(GetVendor.new(params).response)
end

get '/vendors/:vendor_id' do
  json(GetVendor.new(params).response)
end

get '/products' do
  json(GetProducts.new(params).response)
end

get '/' do
  json(GetRoot.new(params).response)
end

